src/AdoNetCore.AseClient/Internal/FormatItem.cs

```diff
@@ -278,5 +279,137 @@ public void WriteForParameter(Stream stream, Encoding enc, TokenType srcTokenTyp
             //locale
             stream.WriteByte(0);
         }
+
+        public string GetDataTypeName()
+        {
+            switch (DataType)
+            {
+                case TdsDataType.TDS_FLT4:
+                    return "real";
+                case TdsDataType.TDS_FLT8:
+                    return "float";
+                case TdsDataType.TDS_FLTN:
+                    switch (Length)
+                    {
+                        case 4:
+                            return "real";
+                        case 8:
+                        default:
+                            return "float";
+                    }
+                case TdsDataType.TDS_BIT:
+                    return "bit";
+                case TdsDataType.TDS_INT1:
+                    return "tinyint";
+                case TdsDataType.TDS_INT2:
+                    return "smallint";
+                case TdsDataType.TDS_INT4:
+                    return "int";
+                case TdsDataType.TDS_INT8:
+                    return "bigint";
+                case TdsDataType.TDS_INTN:
+                    switch (Length)
+                    {
+                        case 1:
+                            return "tinyint";
+                        case 2:
+                            return "smallint";
+                        case 4:
+                            return "int";
+                        case 8:
+                        default:
+                            return "bigint";
+                    }
+                case TdsDataType.TDS_UINT2:
+                    return "unsigned smallint";
+                case TdsDataType.TDS_UINT4:
+                    return "unsigned int";
+                case TdsDataType.TDS_UINT8:
+                    return "unsigned bigint";
+                case TdsDataType.TDS_UINTN:
+                    switch (Length)
+                    {
+                        case 1:
+                            return "unsigned tinyint";
```

Comment:
Is this even a thing?

Reply:
Good spot, that should never get hit -- the server will return a `tinyint`